bin/move-node.js

```diff
@@ -1,16 +1,62 @@
 #!/usr/bin/env node
+const [,, toNode] = process.argv;
 
 const { moveNode, syncShards } = require('../src/move-node');
 const { removeNode } = require('../src/remove-node');
 
+const parseNodeMapping = (input) => {
+  if (!input) {
+    // Single node migration with unspecified node
+    return;
+  }
+  try {
+    // Multi-node migration with specified node mapping
+    // Example input: '{"oldNode":"newNode"}'
+    return JSON.parse(input);
+  } catch (err) {
+    // Single node migration with specified node
+    console.warn('Failed to parse node mapping JSON. Defaulting to single node mapping');
+    return input;
+  }
+};
+
+const getStdin = () => {
+  return new Promise((resolve) => {
+    if (process.stdin.isTTY) {
+      return resolve('');
+    }
+    let data = '';
+    process.stdin.setEncoding('utf8');
+    process.stdin.on('data', chunk => data += chunk);
+    process.stdin.on('end', () => resolve(data));
```

Comment:
We would normally want an error handler here as well. And reject when an error occurs.